lib/ansible/modules/cloud/azure/azure_rm_containerinstance.py

```diff
@@ -280,8 +275,10 @@ def exec_module(self, **kwargs):
         response = None
         results = dict()
 
-        self.mgmt_client = self.get_mgmt_svc_client(ContainerInstanceManagementClient,
-                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+        self.client = self.get_mgmt_svc_client(ContainerInstanceManagementClient)
+
+        # since this client hasn't been upgraded to expose models directly off the OperationClass, fish them out
+        self.cgmodels = self.client.models(self.client.container_groups.api_version)
```

Comment:
I would prefer
```python
self.cgmodels = self.client.container_groups.models
```
like you did for NSG. And yes, this requires a new release of `azure-mgmt-containerinstance`, but I can do that today. Eventually all new packages released since 2 months have the `models` attribute in the operation_group, it's just that when I have no update I didn't release a new package, but I prefer to do that, and not force you to handle this manually with hack.

Reply:
@lmazuel yes please- I hadn't updated in awhile, so a number of things didn't support `models` on operations (which is why I made this ugly hack). When I updated last night, I realized it was being implemented everywhere and tried to get rid of the hack, but noticed that it's still not everywhere yet. :) Would love to remove the hack by 2.5 release time...

Reply:
azure-mgmt-containerinstance 0.3.0 is out, `models` is on the operation group.
https://pypi.python.org/pypi/azure-mgmt-containerinstance/0.3.0

This is a new Api version, but I was able to record the tests I have without changing any code, so should be fine.